tests/unit/events/test_event_listener.py

```diff
@@ -168,22 +168,18 @@ def event_handler(event: BaseEvent):
         mock_event_listener_driver.publish_event.assert_called_once_with({"event": mock_event.to_dict()})
 
     def test_context_manager(self):
```

Comment:
I would like to see a test that shows event listeners are not propagated to new threads or thread pool jobs without the `with_context` util.